src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -264,11 +265,33 @@ public void resetStartsAndStops() {
         assertNotNull(p.getProperty(DisableConcurrentBuildsJobProperty.class));
         assertNull(p.getProperty(BuildDiscarderProperty.class));
 
-        JobPropertyTrackerAction action3 = b3.getAction(JobPropertyTrackerAction.class);
+        JobPropertyTrackerAction action3 = p.getAction(JobPropertyTrackerAction.class);
         assertNotNull(action3);
         assertTrue(action3.getJobPropertyDescriptors().isEmpty());
     }
 
+    @Issue("JENKINS-44848")
+    @LocalData
+    @Test
+    public void trackerPropertyUpgrade() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("trackerPropertyUpgrade", WorkflowJob.class);
+        assertNotNull(p);
+        WorkflowRun b1 = p.getLastBuild();
+        assertNotNull(b1);
+        assertNotNull(p.getProperty(BuildDiscarderProperty.class));
+        assertNull(p.getAction(JobPropertyTrackerAction.class));
+
+        p.setDefinition(new CpsFlowDefinition("properties([disableConcurrentBuilds()])", true));
+
+        r.buildAndAssertSuccess(p);
+
+        assertNull(p.getProperty(BuildDiscarderProperty.class));
```

Comment:
Hmm. Clarifying - the no `JobPropertyTrackerAction` present but a `JobProperty` present case is covered in `onlyRemoveJenkinsfileProperties` - the first run has no `properties` step and so is not setting `JobPropertyTrackerAction` at all. I've cleaned up the test a bit to verify the absence of the action after that run. And I think all the other tests in `JobPropertyStepTest` end up covering the no-completed-builds scenario?